### PR TITLE
#860 - Add code splitting

### LINKED ISSUE #860: Feat: Add Proper Code splitting
### Describe the feature

Code splitting is required for websites to load the components only when it is required. Currently in the website all the components get loaded at the beginning, thus reducing the optimisation of the website.

Hence, I will like to add proper code splitting that can be achieved with React Lazy, suspense with proper fallback

### Add ScreenShots

image
as you can see that just the the first loading of website, all the components get loaded, making the website slow.

### Record

- [X] I agree to follow this project's Code of Conduct
- [X] I'm a GSSOC'23 contributor
- [X] I want to work on this issue

### PR DESCRIPTION
**Pull request by Manice18**
## Fixes Issue 

Closes #860 

## Changes proposed
Previously :- 
image
Now (after adding lazy loading) :- 
image
 

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] My change requires changes to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -1,16 +1,6 @@
...
-      #preloader {
-        background:white url("Hourglass.gif") no-repeat center center;
-        background-size: 10%;
-        height: 100vh;
-        width: 100%;
-        position: fixed;
-        z-index: 999999;
-      }
-      </style>
     <meta charset="utf-8" />
```
Comment (MonalikaPatnaik) [inline public/index.html:4]: Why you have removed the code for preloader?

File: public/index.html [hunk 2]
```diff
@@ -30,20 +20,12 @@
     <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.2/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-EVSTQN3/azprG1Anm3QDgpJLIm9Nao0Yz1ztcQTwFspd3yD65VohhpuuCOmLASjC" crossorigin="anonymous">
     <script type="text/javascript"> window.$crisp=[];window.CRISP_WEBSITE_ID="e79efdd2-abee-4a1e-b868-c7929585ebd9";(function(){ d=document;s=d.createElement("script"); s.src="https://client.crisp.chat/l.js"; s.async=1;d.getElementsByTagName("head")[0].appendChild(s);})(); </script>
   </head>
-  <body class="dark-theme" onload="myFunction()">
```
Comment (MonalikaPatnaik) [inline public/index.html:33]: why you have removed "onload="myFunction()"

**Comment by MonalikaPatnaik** [comment]
It is giving error in deployment..
image

**Comment by Manice18** [comment]
Oh i am sorry it would be 'loader.css' and not 'Loader.css' 
I removed the preloader because it was clashing with the Suspense fallback loader for the home page, hence I added the same loader to all the pages

**Comment by Manice18** [comment]
@MonalikaPatnaik I will resolve all the conflicts and then add a PR

**Pull request closed without merging**